taxcalc/consumption.py

```diff
@@ -0,0 +1,104 @@
+"""
+Tax-Calculator marginal Consumption class.
+"""
+# CODING-STYLE CHECKS:
+# pep8 --ignore=E402 consumption.py
+# pylint --disable=locally-disabled consumption.py
+# (when importing numpy, add "--extension-pkg-whitelist=numpy" pylint option)
+
+from .parameters import ParametersBase
+from .policy import Policy
+from .records import Records
+
+
+class Consumption(ParametersBase):
+    """
+    Constructor for marginal Consumption class.
+
+    Parameters
+    ----------
+    consumption_dict: dictionary of PARAM:DESCRIPTION pairs
+        dictionary of marginal propensity to consume (MPC) parameters;
+        if None, all MPC parameters are read from DEFAULTS_FILENAME file.
+
+    start_year: integer
+        first calendar year for MPC parameters.
+
+    num_years: integer
+        number of calendar years for which to specify MPC parameter
+        values beginning with start_year.
+
+    Raises
+    ------
+    ValueError:
+        if consumption_dict is neither None nor a dictionary.
+        if num_years is less than one.
+
+    Returns
+    -------
+    class instance: Consumption
+    """
+
+    JSON_START_YEAR = Policy.JSON_START_YEAR
+    DEFAULTS_FILENAME = 'consumption.json'
+    DEFAULT_NUM_YEARS = Policy.DEFAULT_NUM_YEARS
+
+    def __init__(self, consumption_dict=None,
+                 start_year=JSON_START_YEAR,
+                 num_years=DEFAULT_NUM_YEARS):
+        super(Consumption, self).__init__()
+        if consumption_dict is None:
+            self._vals = self._params_dict_from_json_file()
+        elif isinstance(consumption_dict, dict):
+            self._vals = consumption_dict
+        else:
+            raise ValueError('consumption_dict is not None or a dictionary')
+        if num_years < 1:
+            raise ValueError('num_years < 1 in Consumption ctor')
+        self.initialize(start_year, num_years)
+
+    def update_consumption(self, revisions):
+        """
+        Update consumption for given revisions, a dictionary consisting
+        of one or more year:modification dictionaries.
+        For example: {2014: {'_MPC_xxx': [0.2, 0.1]}}
+        NOTE: this method uses the specified revisions to update the
+              DEFAULT MPC parameter values, so use this method just once
+              rather than calling it sequentially in an attempt to update
+              MPC parameters in several steps.
+        """
+        precall_current_year = self.current_year
+        self.set_default_vals()
+        # specify revisions
```

Comment:
Lines 73-76 tripped me up for a second. I didn't understand the need to create `revision_years` as a sorted list before iterating through the dictionary. Then I saw that the desire is to go through the revisions in order year by year. So, since there is already a comment on line 72, I would suggest `# specify revisions in order by year` to give the reader a clue for why the keys are put in a sorted list first.